### DISCUSSION TITLE
Hai i have a doubt , i am seeing an error when i tried to pass random data through CNN , i am sharing the code here:

### DISCUSSION THREAD

**Discussion by tedzz12** [mrdbourke/pytorch-deep-learning#403] [General]
import torch
from torch import nn

class conv_nn6(nn.Module):
  def __init__(self,input:int,hidden:int,output:int):
    super().__init__()
    self.conv_block1 = nn.Sequential(
        nn.Conv2d(
            in_channels=input,
            out_channels=hidden,
            kernel_size=3,
            stride=1,
            padding=1
        ),
        nn.ReLU(),
        nn.Conv2d(
            in_channels=hidden,
            out_channels=hidden,
            kernel_size=3,
            stride=1,
            padding=1
        ),
        nn.ReLU(),
        nn.MaxPool2d(kernel_size=2)
    )

    self.conv_block2 = nn.Sequential(
        nn.Conv2d(
            in_channels=hidden,
            out_channels=hidden,
            kernel_size=3,
            stride=1,
            padding=1
        ),
        nn.ReLU(),
        nn.Conv2d(
            in_channels=hidden,
            out_channels=hidden,
            kernel_size=3,
            stride=1,
            padding=1
        ),
        nn.ReLU(),
        nn.MaxPool2d(kernel_size=2)
    )
      
    self.classifier = nn.Sequential(
        nn.Flatten(),
        nn.Linear(in_features=hidden*7*7, 
                      out_features=output)
      )
        
        
        
        
        
  def forward(self,x:torch.Tensor):
      x = self.conv_block1(x)
      print(f"Shape of conv_block1 {x.shape}")
      x = self.conv_block2(x)
      print(f"Shape of conv_block2 {x.shape}")
      x = self.classifier(x)
      print(f"Shape of classifier {x.shape}")
      return x              model_xv = conv_nn6(input=1,
                     hidden=10,
                     output=10)

model_xv  rand_image = torch.randn(1,28,28)
rand_image

model_xv(rand_image)  

ERROR:->
:----------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
[<ipython-input-70-bf59604247c8>](https://localhost:8080/#) in <cell line: 4>()
      2 rand_image
      3 
----> 4 model_xv(rand_image)

5 frames
[/usr/local/lib/python3.9/dist-packages/torch/nn/modules/linear.py](https://localhost:8080/#) in forward(self, input)
    112 
    113     def forward(self, input: Tensor) -> Tensor:
--> 114         return F.linear(input, self.weight, self.bias)
    115 
    116     def extra_repr(self) -> str:

RuntimeError: mat1 and mat2 shapes cannot be multiplied (10x49 and 490x10)

**Comment by AugustaSousa** [comment]
CNNs usually work with 4 ‘dimensions’ for 2D images (N, C, H, W) or (N, H, W, C). 
The rand_image has (1, 28, 28), I think there was a misinterpretation and it treated the 10 in (10,7,7) as an N, so it would return a shape of (10,10) instead of (1,10).
The `model_xv(rand_image.unsqueeze(dim=0))` will work, (adding an extra dimension to the tensor, which represents the batch size)
(1, 28, 28) -> (1, 1, 28, 28)
But I would like if someone could explain that misinterpretation better...

**Reply by tedzz12** [reply]
Thank you i understand

**Comment by tedzz12** [comment]
i also have another dobt, after  the training of the CNN with FashionMNIST
, how can i connect CNN to the open cv?

**Reply by AugustaSousa** [reply]
I won't be able to answer well now because I'm still learning how to use Opencv (T.T)

**Comment by tedzz12** [comment]
No problem, thanks!
